contrib/pg_tde/documentation/docs/replication.md

```diff
@@ -1,4 +1,257 @@
+
+1. **Install the `pg_tde` extension files.**
+2. **Configure `postgresql.conf` for `pg_tde`:**
+
+    ```ini
+    shared_preload_libraries = 'pg_tde' // Add pg_tde, comma-separated if others exist
+
+    # Example for File Provider:
+    pg_tde_key_provider = 'file'
+    pg_tde_file_keys_path = '/var/lib/postgresql/pg_tde_keys'
+    pg_tde_principal_key_name = 'my_principal_key'
+
+    # Example for Vault Provider:
+    # pg_tde_key_provider = 'vault'
+    # pg_tde_vault_url = 'http://your-vault-server:8200'
+    # pg_tde_vault_token = 's.xxxxxxxxxxxxxx'
+    # pg_tde_vault_mount_path = 'secret/'
+    # pg_tde_principal_key_name = 'my_principal_key_in_vault'
+    ```
+
+3. **Configure `postgresql.conf` for Replication:**
+
+    ```ini
+    wal_level = replica
+    max_wal_senders = 5
+    archive_mode = on # Recommended
+    archive_command = 'cp %p /path/to/archive/%f' # Example
+    ```
+
+4. **Prepare principal Key in Provider:**
+
+    * **File:** Create the directory specified in `pg_tde_file_keys_path`. Create the key file inside this directory, named as per `pg_tde_principal_key_name`.
+
+        ```bash
+        sudo -u postgres mkdir -p /var/lib/postgresql/pg_tde_keys
+        sudo -u postgres chmod 700 /var/lib/postgresql/pg_tde_keys
+        sudo -u postgres dd if=/dev/urandom of=/var/lib/postgresql/pg_tde_keys/my_principal_key bs=32 count=1
+        sudo -u postgres chmod 600 /var/lib/postgresql/pg_tde_keys/my_principal_key
+        ```
+
+    * **Vault:** Ensure the key exists in Vault at the correct path and is accessible.
+5. **Restart PostgreSQL on Primary.**
+6. **Initialize `pg_tde` and Set principal Key (in `psql`):**
+
+    ```sql
+    CREATE EXTENSION pg_tde;
+    -- Add the key definition to pg_tde's catalog
+    -- For File (key_name matches pg_tde_principal_key_name and filename):
+    SELECT pg_tde_principal_key_add('my_principal_key', NULL, 'file');
+    -- For Vault (key_name matches pg_tde_principal_key_name and key name in Vault):
+    -- SELECT pg_tde_principal_key_add('my_principal_key_in_vault', NULL, 'vault');
+
+    -- Set this key as the active principal key
+    -- For File:
+    SELECT pg_tde_set_principal_key('my_principal_key', 'file');
+    -- For Vault:
+    -- SELECT pg_tde_set_principal_key('my_principal_key_in_vault', 'vault');
+
+    -- Set heap_tde as default for new tables (optional)
+    ALTER SYSTEM SET default_table_access_method = 'heap_tde';
+    SELECT pg_reload_conf();
+    ```
+
+7. **Create Replication User & Configure `pg_hba.conf` (standard PostgreSQL steps).**
+8. **Create and Encrypt a Test Table:**
+
+    ```sql
+    -- If default_table_access_method = 'heap_tde'
+    CREATE TABLE encrypted_data (id INT PRIMARY KEY, data TEXT);
+    -- Or explicitly:
+    -- CREATE TABLE encrypted_data (id INT PRIMARY KEY, data TEXT) USING heap_tde;
+
+    INSERT INTO encrypted_data VALUES (1, 'Secret on primary');
+    ```
+
+### B. Standby Server Setup**
+
+1. **Install `pg_tde` extension files.**
```

Comment:
The first step should be "create the replica using basebackup or something", please check/try how replication setup works with postgres.